### PR TITLE
Merge GoalPage with GoalPreview

### LINKED ISSUE #1430: Merge GoalPage with GoalPreview
can't it be a component instead of a copy?

_Originally posted by @9teen90nine in https://github.com/taskany-inc/issues/pull/1410#discussion_r1282974818_

### LINKED ISSUE #1718: GoalPreview have two scrollable containers with extra padding
Image

Open goal preview
Click by comment time
Anchor be scrolling to top of container
Next scroll action doesn't works correcty

### PR DESCRIPTION
**Pull request by DenisVorop**
## PR includes
- [x] Refactor

## Related issues
Resolve #1430 
Resolve #1718 

## QA Instructions, Screenshots, Recordings

related to the #1718 

https://github.com/taskany-inc/issues/assets/95316053/51062f18-a541-4c97-a8fe-c19317c965f0

in #1715 block is scrolling horizontally because a long line of text is not being automatically wrapped onto the next line by default, causing the content to extend beyond the visible area.

### CODE REVIEW

File: src/components/GoalHeader.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+import styled from 'styled-components';
+
+import { GoalByIdReturnType } from '../../trpc/inferredTypes';
+
+import StateSwitch from './StateSwitch';
+import { State } from './State';
+import { IssueStats } from './IssueStats/IssueStats';
+import { IssueTitle } from './IssueTitle';
+
+const IssueHeader = styled.div`
```
Comment (awinogradov) [inline src/components/GoalHeader.tsx:12]: Must be `StyledGoalHeader`

https://github.com/taskany-inc/issues/blob/main/docs/patterns.md

File: src/components/GoalHeader.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    ${({ align }) =>
+        align === 'right' &&
+        `
+            display: grid;
+            justify-items: end;
+            align-content: space-between;
+        `}
+`;
+
+type IssueTitleProps = ComponentProps<typeof IssueTitle>;
```
Comment (awinogradov) [inline src/components/GoalHeader.tsx:36]:
``` ts
interface GoalHeaderProps extends ComponentProps<typeof IssueTitle> {
   //...
}
```

File: src/components/GoalInfo/GoalInfo.tsx [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+        display: block;
+    }
+`;
+
+const tagsLimit = 5;
+
+type Tag = NonNullable<GoalByIdReturnType>['tags'][number];
+type UserComboBoxProps = ComponentProps<typeof UserComboBox>;
+
+type GoalInfoProps = {
```
Comment (awinogradov) [inline src/components/GoalInfo/GoalInfo.tsx:48]: interface

File: src/components/GoalInfo/GoalInfo.tsx [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+    participants,
+    onGoalAssigneeChange,
+    onGoalParticipantRemove,
+    onGoalParticipantAdd,
+    onGoalTagRemove,
+    onGoalTagAdd,
+    onGoalTransfer,
+}) => {
+    return (
+        <div>
```
Comment (awinogradov) [inline src/components/GoalInfo/GoalInfo.tsx:70]: Does this is the part of GoalPage layout? Like this https://github.com/taskany-inc/issues/pull/1745/files#diff-b842f4ad5ef9f56007262c1d647b0bdb1d7bbae4275dda8b2f2ab31f81e7c869R257

File: src/components/GoalInfo/GoalInfo.tsx [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+                            </StyledInlineUserInput>
+                        )}
+                    />
+                ) : (
+                    <UserBadge user={goal.owner?.user} />
+                )}
+            </IssueMeta>
+
+            <IssueMeta title={tr('Participants')}>
+                {goal.participants?.map(({ user }) => (
```
Comment (awinogradov) [inline src/components/GoalInfo/GoalInfo.tsx:95]: Must be merged with #1762

File: src/components/GoalContent.tsx [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+const ImageFullScreen = dynamic(() => import('./ImageFullScreen'));
+
+type GoalCriteriaProps = ComponentProps<typeof GoalCriteria>;
+type CommentViewProps = ComponentProps<typeof CommentView>;
+type CommentCreateFormProps = ComponentProps<typeof CommentCreateForm>;
+type GoalDependencyListByKindProps = ComponentProps<typeof GoalDependencyListByKind>;
+
+type Comment = NonNullable<GoalByIdReturnType>['comments'][number];
+
+type GoalContentProps = {
```
Comment (awinogradov) [inline src/components/GoalContent.tsx:31]: interface

Comment (awinogradov) [reply src/components/GoalContent.tsx:31]: extends, Pick — anyway with GoalCriteriaProps

File: src/components/GoalContent.tsx [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+                                onSubmit={onGoalCommentSubmit(value)}
+                                onReactionToggle={onGoalCommentReactionToggle?.(value.id)}
+                                onDelete={onGoalCommentDelete?.(value.id)}
+                            />
+                        )}
+                    />
+                ))}
+
+                {nullable(goal, (g) =>
+                    nullable(g._isEditable, () => (
```
Comment (awinogradov) [inline src/components/GoalContent.tsx:187]: `g?._isEditable`

**Review by awinogradov — CHANGES_REQUESTED** [review]

File: src/components/GoalContent.tsx (hunk 7 above)
Comment (DenisVorop) [reply src/components/GoalContent.tsx]: GoalEditForm & GoalDeleteModal don't accept goal prop when its undefined, this check is necessary.

File: src/components/GoalActivityFeed.tsx [hunk 8]
```diff
@@ -0,0 +1,191 @@
...
+import { GoalDependencyListByKind } from './GoalDependencyList/GoalDependencyList';
+import { GoalCriteria } from './GoalCriteria/GoalCriteria';
+import { GoalActivity } from './GoalActivity';
+import { AddCriteriaForm } from './CriteriaForm/CriteriaForm';
+
+const ModalOnEvent = dynamic(() => import('./ModalOnEvent'));
+const GoalEditForm = dynamic(() => import('./GoalEditForm/GoalEditForm'));
+const CommentCreateForm = dynamic(() => import('./CommentCreateForm/CommentCreateForm'));
+
+type GoalActivityFeedProps = {
```
Comment (awinogradov) [inline src/components/GoalActivityFeed.tsx]: interface

File: src/components/GoalActivityFeed.tsx [hunk 9]
```diff
@@ -0,0 +1,191 @@
...
+    goal?: GoalByIdReturnType;
+    shortId?: string;
+    activityId?: string;
+
+    onGoalCriteriaClick?: ComponentProps<typeof GoalCriteria>['onClick'];
+    onGoalDependencyClick?: ComponentProps<typeof GoalDependencyListByKind>['onClick'];
+    onDelete?: () => void;
```
Comment (awinogradov) [inline src/components/GoalActivityFeed.tsx:29]: on delete what?

File: src/components/GoalActivityFeed.tsx [hunk 10]
```diff
@@ -0,0 +1,191 @@
...
+        const onGoalCommentSubmit = useCallback(
+            (value: NonNullable<GoalByIdReturnType>['comments'][number]) => {
+                return value.activity?.id === activityId ? onGoalCommentUpdate(value.id) : undefined;
+            },
+            [onGoalCommentUpdate, activityId],
+        );
+
+        return (
+            <>
+                {nullable(goal, ({ _activityFeed, id, goalAchiveCriteria, _relations, project, _isEditable }) => (
```
Comment (awinogradov) [inline src/components/GoalActivityFeed.tsx:82]: This logic must be in parent component. U can check it in parent and make goal prop required to render this component

File: src/components/GoalActivityFeed.tsx [hunk 11]
```diff
@@ -0,0 +1,191 @@
...
+import { AddCriteriaForm } from './CriteriaForm/CriteriaForm';
+
+const ModalOnEvent = dynamic(() => import('./ModalOnEvent'));
+const GoalEditForm = dynamic(() => import('./GoalEditForm/GoalEditForm'));
+const CommentCreateForm = dynamic(() => import('./CommentCreateForm/CommentCreateForm'));
+
+type GoalActivityFeedProps = {
+    goal?: GoalByIdReturnType;
+    shortId?: string;
+    activityId?: string;
```
Comment (awinogradov) [inline src/components/GoalActivityFeed.tsx:25]: `usePageContext`

File: src/components/GoalActivityFeed.tsx [hunk 12]
```diff
@@ -0,0 +1,191 @@
...
+                                                    onSubmit={props.onSubmit}
+                                                    validityData={props.validityData}
+                                                />
+                                            ))
+                                        }
+                                    />
+                                ))}
+
+                                {_relations.map((deps) =>
+                                    nullable(deps.goals.length || _isEditable, () => (
```
Comment (awinogradov) [inline src/components/GoalActivityFeed.tsx:112]: I'm not sure this logic is ok.

image

File: src/components/GoalActivityFeed.tsx [hunk 13]
```diff
@@ -0,0 +1,191 @@
...
+                                            ))}
+                                        </GoalDependencyListByKind>
+                                    )),
+                                )}
+
+                                {nullable(lastStateComment, (value) => (
+                                    <CommentView
+                                        pin
+                                        id={value.id}
+                                        author={value.activity?.user}
```
Comment (awinogradov) [inline src/components/GoalActivityFeed.tsx:138]: getUser? cc @LamaEats

File: src/components/GoalActivityFeed.tsx [hunk 14]
```diff
@@ -0,0 +1,191 @@
...
+                                description={commentDraft?.description}
+                                onSubmit={onGoalCommentCreate}
+                                onCancel={onGoalCommentCancel}
+                                onChange={onGoalCommentChange}
+                            />
+                        }
+                        renderCommentItem={(value) => (
+                            <CommentView
+                                id={value.id}
+                                author={value.activity?.user}
```
Comment (awinogradov) [inline src/components/GoalActivityFeed.tsx:163]: getUser?

File: src/components/GoalHeader.tsx [hunk 15]
```diff
@@ -0,0 +1,95 @@
...
+    children?: ReactNode;
+
+    onGoalStateChange?: ComponentProps<typeof StateSwitch>['onClick'];
+}
+
+export const GoalHeader: FC<GoalHeaderProps> = ({
+    goal,
+    actions,
+    children,
+
```
Comment (awinogradov) [inline src/components/GoalHeader.tsx:50]: ?

File: src/components/GoalHeader.tsx [hunk 16]
```diff
@@ -0,0 +1,95 @@
...
+    href,
+    size,
+
```
Comment (awinogradov) [inline src/components/GoalHeader.tsx:53]: ?

File: src/components/GoalSidebar/GoalSidebar.tsx [hunk 17]
```diff
@@ -0,0 +1,180 @@
...
+    &:hover ${StyledIconEditOutline} {
+        display: block;
+    }
+`;
+
+const tagsLimit = 5;
+
+interface GoalSidebarProps {
+    goal: NonNullable<GoalByIdReturnType>;
+    participants: string[];
```
Comment (awinogradov) [inline src/components/GoalSidebar/GoalSidebar.tsx:47]: U have goal.participants

File: src/components/GoalSidebar/GoalSidebar.tsx [hunk 18]
```diff
@@ -0,0 +1,180 @@
...
+    participants,
+    onGoalAssigneeChange,
+    onGoalParticipantRemove,
+    onGoalParticipantAdd,
+    onGoalTagRemove,
+    onGoalTagAdd,
+    onGoalTransfer,
+}) => {
+    return (
+        <div>
```
Comment (awinogradov) [inline src/components/GoalSidebar/GoalSidebar.tsx:67]: Does this is part of parent layout?

File: src/components/GoalSidebar/GoalSidebar.tsx [hunk 19]
```diff
@@ -0,0 +1,180 @@
...
+                        onChange={onGoalAssigneeChange}
+                        renderTrigger={(props) => (
+                            <StyledInlineUserInput>
+                                <UserBadge user={goal.owner?.user} />
+                                <StyledIconEditOutline size="xxs" onClick={props.onClick} />
+                            </StyledInlineUserInput>
+                        )}
+                    />
+                ) : (
+                    <UserBadge user={goal.owner?.user} />
```
Comment (awinogradov) [inline src/components/GoalSidebar/GoalSidebar.tsx:87]: getUser?

File: src/components/GoalSidebar/GoalSidebar.tsx [hunk 20]
```diff
@@ -0,0 +1,180 @@
...
+                )}
+            </IssueMeta>
+
+            {nullable(goal._isEditable || goal.participants.length, () => (
+                <IssueMeta title={tr('Participants')}>
+                    {goal.participants?.map(({ user }) => (
+                        <UserBadge
+                            key={user?.activityId}
+                            user={user}
```
Comment (awinogradov) [inline src/components/GoalSidebar/GoalSidebar.tsx:96]: getUser?

File: src/components/GoalSidebar/GoalSidebar.tsx [hunk 21]
```diff
@@ -0,0 +1,180 @@
...
+                                        onClick={props.onClick}
+                                    />
+                                )}
+                            />
+                        </StyledInlineInput>
+                    ))}
+                </IssueMeta>
+            ))}
+
+            {nullable(goal._isEditable || goal.tags.length, () => (
```
Comment (awinogradov) [inline src/components/GoalSidebar/GoalSidebar.tsx:123]: I'm not sure

**Review by awinogradov — CHANGES_REQUESTED** [review]

File: src/components/GoalContent.tsx (hunk 7 above)
Comment (awinogradov) [reply src/components/GoalContent.tsx]: goal prop must be required

**Comment by awinogradov** [comment]
https://github.com/taskany-inc/issues/pull/1723/files#diff-67f44f7eacc06dc6b9b67aeafac53c731d3e2b887eb0d9cf4f71405634cf92cdR230 this is what I mean in comments with `getUser`

File: src/components/GoalActivityFeed.tsx (hunk 13 above)
Comment (DenisVorop) [reply src/components/GoalActivityFeed.tsx]: author is the User model, getUserName using by internal component

in other cases its the same

File: src/components/GoalActivityFeed.tsx (hunk 12 above)
Comment (DenisVorop) [reply src/components/GoalActivityFeed.tsx]: this logic is equivalent to the previous one, but its easier to understand

**Review by awinogradov — APPROVED** [review]

**Review by 9teen90nine — APPROVED** [review]

**Pull request merged**